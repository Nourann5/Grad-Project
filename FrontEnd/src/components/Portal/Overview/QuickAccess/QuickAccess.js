import React from 'react'
import { Link } from 'react-router-dom'
import styles from './QuickAccess.module.css'

import {ReactComponent as OverviewIcon} from 'assets/icons/overview.svg'
import {ReactComponent as CoursePlanPortalIcon} from 'assets/icons/coursePlanPortal.svg'
import {ReactComponent as ExamsPortalIcon} from 'assets/icons/examsPortal.svg'
import {ReactComponent as GroupsPortalIcon} from 'assets/icons/groupsPortal.svg'
import {ReactComponent as MaterialsPortalIcon} from 'assets/icons/materialsPortal.svg'
import {ReactComponent as MembersPortalIcon} from 'assets/icons/membersPortal.svg'
import {ReactComponent as ReportsPortalIcon} from 'assets/icons/reportsPortal.svg'
import {ReactComponent as TodolistPortalIcon} from 'assets/icons/todolistPortal.svg'
import {ReactComponent as ChatPortalIcon} from 'assets/icons/chatPortal.svg'
import {ReactComponent as SettingsPortalIcon} from 'assets/icons/settingsPortal.svg'
import { Col, Row } from 'react-bootstrap'
function QuickAccess() {
  return (
    <div className={styles['quick-access']}>
        <h2 className={styles['quick-access__title']}>
            Quick Access
        </h2>
        <div className={styles['quick-access__items-wrapper']}>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <OverviewIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Add Test</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <CoursePlanPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>All Test</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <ExamsPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Add Group</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <GroupsPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>All Groups</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <MaterialsPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>All Materials</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <MembersPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Add Member</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <ReportsPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Reports</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <TodolistPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Todolist</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <ChatPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Chats</span>
                        </div>
                    </div>
                    <div className={styles['quick-access__item']}>
                        <div className={styles['quick-access__item-content']}>
                        <Link to='/' className={styles['quick-access__link']}></Link>
                            <div className={styles['quick-access__item-icon-wrapper']}>
                                <SettingsPortalIcon className={styles['quick-access__item-icon']}/>
                            </div>
                            <span className={styles['quick-access__item-value']}>Settings</span>
                        </div>
                    </div>
        </div>
    </div>
  )
}

export default QuickAccess