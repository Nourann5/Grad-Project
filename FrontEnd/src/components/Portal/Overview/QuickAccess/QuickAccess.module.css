.quick-access{
    margin-bottom: 30px;
}
.quick-access__title{
    font-size: 30px;
    font-family: 'PoppinsBold';
    color: var(--main-color);
    margin-bottom: 20px;
}
.quick-access__items-wrapper{
    display: grid;
    grid-template-columns: repeat(7,1fr);
    grid-gap: 20px;
}
.quick-access__item{

}
.quick-access__link{
    position: absolute;
    z-index: 1;
    top: 0;
    left: 0;
    right: 0;
    border: 0;
    width: 100%;
    height: 100%;
}
.quick-access__item-content{
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    width: max-content;
    margin: auto;
    transition:var(--transition);
}
.quick-access__item-icon-wrapper{
    background-color: white;
    border-radius: 10px;
    padding: 25px 30px;
    margin-bottom: 10px;
    box-shadow: 0 0px 4px 1px rgb(0 0 0 / 10%);
    transition:var(--transition);
}
.quick-access__item-content:hover{
    box-shadow: none;
    transform: translateY(-5px);

}
.quick-access__item-value{

}
.quick-access__item:nth-of-type(1) .quick-access__item-icon-wrapper{
    background-color: #8ecae6;
}
.quick-access__item:nth-of-type(2) .quick-access__item-icon-wrapper{
    background-color: #219ebc;
}
.quick-access__item:nth-of-type(3) .quick-access__item-icon-wrapper{
    background-color: #023047;
}
.quick-access__item:nth-of-type(4) .quick-access__item-icon-wrapper{
    background-color: #fb8500;
}
.quick-access__item:nth-of-type(5) .quick-access__item-icon-wrapper{
    background-color: #A4668B;
    /* background-color: #941b0c; */
}
.quick-access__item:nth-of-type(6) .quick-access__item-icon-wrapper{
    background-color: #a3b18a;
}
.quick-access__item:nth-of-type(7) .quick-access__item-icon-wrapper{
    background-color: #8ac926;
}
.quick-access__item:nth-of-type(8) .quick-access__item-icon-wrapper{
    background-color: #7678ed;
}
.quick-access__item:nth-of-type(9) .quick-access__item-icon-wrapper{
    /* background-color: #EDB0B8; */
    background-color: #b75d69;
}
.quick-access__item:nth-of-type(10) .quick-access__item-icon-wrapper{
    background-color: #dab49d;
}
.quick-access__item:nth-of-type(11) .quick-access__item-icon-wrapper{
    background-color: #861657;
}
.quick-access__item:nth-of-type(12) .quick-access__item-icon-wrapper{
    /* background-color: #ffc6ff; */
    background-color: #D78D97;
}
.quick-access__item:nth-of-type(1) .quick-access__item-icon path{
    stroke:white
}
.quick-access__item-icon path{
    fill:white
}
.quick-access__item-icon ellipse{
    fill:white
}
.quick-access__item-icon circle{
    fill:white
}