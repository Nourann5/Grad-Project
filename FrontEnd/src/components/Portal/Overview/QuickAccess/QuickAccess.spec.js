import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import QuickAccess from './QuickAccess'

describe('Test the Portal Quick Access', () => { 
    it('heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <QuickAccess/>
            </BrowserRouter>
        )
        expect(tree).toMatchSnapshot()
    })
 })