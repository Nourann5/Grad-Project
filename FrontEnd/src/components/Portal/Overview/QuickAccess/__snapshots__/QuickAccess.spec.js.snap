// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Portal Quick Access heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quick-access"
      >
        <h2
          class="quick-access__title"
        >
          Quick Access
        </h2>
        <div
          class="quick-access__items-wrapper"
        >
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  overview.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Add Test
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  coursePlanPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                All Test
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  examsPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Add Group
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  groupsPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                All Groups
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  materialsPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                All Materials
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  membersPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Add Member
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  reportsPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Reports
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  todolistPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Todolist
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  chatPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Chats
              </span>
            </div>
          </div>
          <div
            class="quick-access__item"
          >
            <div
              class="quick-access__item-content"
            >
              <a
                class="quick-access__link"
                href="/"
              />
              <div
                class="quick-access__item-icon-wrapper"
              >
                <svg
                  class="quick-access__item-icon"
                >
                  settingsPortal.svg
                </svg>
              </div>
              <span
                class="quick-access__item-value"
              >
                Settings
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quick-access"
    >
      <h2
        class="quick-access__title"
      >
        Quick Access
      </h2>
      <div
        class="quick-access__items-wrapper"
      >
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                overview.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Add Test
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                coursePlanPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              All Test
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                examsPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Add Group
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                groupsPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              All Groups
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                materialsPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              All Materials
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                membersPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Add Member
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                reportsPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Reports
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                todolistPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Todolist
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                chatPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Chats
            </span>
          </div>
        </div>
        <div
          class="quick-access__item"
        >
          <div
            class="quick-access__item-content"
          >
            <a
              class="quick-access__link"
              href="/"
            />
            <div
              class="quick-access__item-icon-wrapper"
            >
              <svg
                class="quick-access__item-icon"
              >
                settingsPortal.svg
              </svg>
            </div>
            <span
              class="quick-access__item-value"
            >
              Settings
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
